src/Bounds.cpp

```diff
@@ -653,7 +653,7 @@ class Bounds : public IRVisitor {
                 interval.max = b.max - make_one(t);
             } else if (b.max.type().is_uint()) {
                 // if b.max = 0 then result is [0, 0], else [0, b.max - 1]
-                interval.max = select(is_zero(b.max), make_zero(t), b.max - make_one(t));
+                interval.max = select(b.max == make_zero(t), make_zero(t), b.max - make_one(t));
```

Comment:
I had to think a bit before realizing the nature of the change. Maybe we should rename `is_zero` and friends to make this more clear?

Reply:
Yeah... maybe is_const_zero?

Reply:
(and is_const_one)

Reply:
SGTM